### PR TITLE
Deprecate unused testing cleanup mechanisms

### LINKED ISSUE #14329: Deprecate unused testing cleanup mechanisms
The class `matplotlib.testing.decorators.CleanupTestCase` and the decorator `matplotlib.testing.decorators.cleanup` are not used in the code base. We have an automatic cleanup via an auto-used fixture instead.

Can we deprecate them or are they considered part of the user-facing API for testing?

### PR DESCRIPTION
**Pull request by SidharthBansal**
## PR Summary

The class matplotlib.testing.decorators.CleanupTestCase and the decorator matplotlib.testing.decorators.cleanup are not used in the code base. We have an automatic cleanup via an auto-used fixture instead. This PR removes them.
Fixes #14329

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by SidharthBansal** [comment]
@jklymak @story645 kindly review

File: lib/matplotlib/testing/decorators.py [hunk 1]
```diff
@@ -35,58 +35,6 @@ def _cleanup_cm():
         plt.close("all")
 
 
-class CleanupTestCase(unittest.TestCase):
```
Comment (jklymak) [inline lib/matplotlib/testing/decorators.py:38]: This isn't a deprecation, you have just removed these. 

What is the motivation for deprecating these?  Just general cleanup?

**Comment by jklymak** [comment]
I take https://github.com/matplotlib/matplotlib/issues/14329#issuecomment-496007079 to mean that @annzter would explicitly like to keep these methods.

**Comment by SidharthBansal** [comment]
I asked in chatroom shall I remove these @story645 asked to raise a pr. I
think, there might be some miscommunication. In case, we need to keep these
we can close the PR and issue both. As opened issue creates a problem. New
contributors often submit a pr to unclosed issue without checking it's need
until labels like `wont-fix` etc. are posted on them.
Sorry, I am currently learning deprecations here. Earlier, in deprecations
at other orgs, we remove that code OR use another method/class that is more
robust instead of the current one. So, there is tech stack difference too.
So, I am currently learning about decorators and wrappings. :-)

**Comment by SidharthBansal** [comment]
Small typo @anntzer instead of @Anntzer. Sorry @Anntzer for disturbing you.

**Comment by anntzer** [comment]
I think these can/should stay for now.

@SidharthBansal Matplotlib is a very widely used library, and effectively anything that we expose is likely used by tons of people regardless of whether we intended it for external use or not.  Thereforce we try not to remove stuff unless there's a clear reason to ("clear" being something that you can only evaluate with some experience...) and even then, we need a deprecation period where the thing stays but third-party users who rely on it get a warning that the API is going away.

In this specific case, I think this API has a legitimate use case and is self-contained enough that it does not cause undue maintenance burden by staying.

**Comment by SidharthBansal** [comment]
> we need a deprecation period where the thing stays but third-party users
who rely on it get a warning that the API is going away.

That is a new thing I learnt today. Thanks for thorough explanation. So,
shall I deprecate it OR we are aiming at closing the PR? If we are willing
to close the PR, my humble suggestion will be to attach `wont-fix` type
label(whatsoever used here) at respective issue so that no one else will
attempt to submit a pr for it until we necessarily need it or simply close
the issue.

Thanks for helping me. You all are very helpful :-)

**Comment by anntzer** [comment]
My opinion is to close as wontfix, but again let's give @timhoffm time to give his opinion as well.  Or sometimes issues just need more discussion :)

**Comment by jklymak** [comment]
I think we can proactively close this.  If @timhoffm wants more discussion, he can re-open.

**Comment by SidharthBansal** [comment]
Agreed!

**Comment by timhoffm** [comment]
@anntzer I'm fine with keeping if you think it's worth.

**Pull request closed without merging**